packages/flutter/lib/src/gestures/converter.dart

```diff
@@ -210,31 +206,21 @@ class PointerEventConverter {
               radiusMax: radiusMax,
               embedderId: datum.embedderId,
             );
-            break;
-        }
-      } else {
-        switch (datum.signalKind!) {
-          case ui.PointerSignalKind.scroll:
-            final Offset scrollDelta =
-                Offset(datum.scrollDeltaX, datum.scrollDeltaY) / devicePixelRatio;
-            yield PointerScrollEvent(
-              timeStamp: timeStamp,
-              kind: kind,
-              device: datum.device,
-              position: position,
-              scrollDelta: scrollDelta,
-              embedderId: datum.embedderId,
-            );
-            break;
-          case ui.PointerSignalKind.none:
-            assert(false); // This branch should already have 'none' filtered out.
-            break;
-          case ui.PointerSignalKind.unknown:
-            // Ignore unknown signals.
-            break;
         }
       }
-    }
+      // This branch should already have 'none' and 'unknown' filtered out.
+      assert(datum.signalKind == ui.PointerSignalKind.scroll);
```

Comment:
actually the reason we use switch here is so that if we ever add a new kind of signal, the linter will flag this as some code to look at

Reply:
Yeah... I'm changing it back to the switch, but will be throwing something to avoid the compiler thinking that we actually want to return anything there - since we're filtering it out above it will be safe.